src/js/leaflet.storage.js

```diff
@@ -683,20 +700,33 @@ L.Storage.Map.include({
                 layer;
             if (layerId) layer = map.datalayers[layerId];
             if (fileInput.files.length) {
-                if (layer) {
-                    layer.importFromFiles(fileInput.files, type);
-                } else {
-                    for (var i = 0, f; f = fileInput.files[i]; i++) {
-                        layer = this.createDataLayer({name: f.name});
-                        layer.importFromFile(f, type);
+                var file;
+                for (var i = 0, f; f = fileInput.files[i]; i++) {
+                    file = fileInput.files[i];
+                    if (L.Util.detectFileType(file) === 'umap' || type === 'umap') {
+                        this.importFromFile(file, 'umap');
+                    } else {
+                        var importLayer = layer;
+                        if (!layer) {
+                            importLayer = this.createDataLayer({name: f.name});
+                        }
+                        importLayer.importFromFile(f, type);
                     }
                 }
             } else {
-                if (!layer) layer = this.createDataLayer();
                 if (!type) return L.S.fire('ui:alert', {content: L._('Please choose a format'), level: 'error'});
-                else if (rawInput.value) layer.importRaw(rawInput.value, type);
-                else if (urlInput.value) layer.importFromUrl(urlInput.value, type);
-                else if (presetSelect.selectedIndex > 0) layer.importFromUrl(presetSelect[presetSelect.selectedIndex].value, type);
+                if (rawInput.value && type === 'umap') {
```

Comment:
Maybe you can move the try/catch to the `importRaw` method, so we keep this piece of code consistent and more readable.